include/ck/wrapper/layout.hpp

```diff
@@ -327,19 +383,51 @@ struct Layout
      *
      * \return Shape.
      */
-    __host__ __device__ constexpr Shape GetShape() const { return shape_; }
+    __host__ __device__ constexpr auto& GetShape() const { return shape_; }
```

Comment:
Why changing to `auto`? imo, it's making it less readable. If you want auto here, I would suggest using `const auto&` to be more explicit.